### DISCUSSION TITLE
Unable to Successfully Close Material Property Setting in Specific Time Period

### DISCUSSION THREAD

**Discussion by ggb074113** [idaholab/moose#29541] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello! I have utilized the timeperiod functionality within the control module to dynamically set the material properties within a certain block. Specifically, I aim for the same material within the same block to have three different property values across three distinct time periods. Therefore, I have configured each timeperiod to enable the corresponding material property while attempting to disable the other two.
However, during actual simulation runs, I have observed that one of the material properties remains consistently enabled. Even when I explicitly specify to disable it within a certain timeperiod, it stays enabled.
![Ubuntu 64 位-2024-12-14-19-37-34](https://github.com/user-attachments/assets/0767966a-aed1-442c-a13b-12604fc2ae0d)
![Ubuntu 64 位-2024-12-14-19-37-34](https://github.com/user-attachments/assets/02aa73df-d229-4c4d-8f71-c3e256d4cf1e)
![Ubuntu 64 位-2024-12-14-19-39-50](https://github.com/user-attachments/assets/27c1f965-9bb1-408f-93ce-1f654c5da97b)
![Ubuntu 64 位-2024-12-14-19-40-00](https://github.com/user-attachments/assets/21540bff-896e-4cf0-8143-1e13e14aada7)

**Comment by ggb074113** [comment]
When I try to close mat5， mat5 still can't close![Ubuntu 64 位-2024-12-14-19-42-54](https://github.com/user-attachments/assets/785c4081-41ad-409c-afff-23dfd357a513)

**Reply by GiudGiud** [reply]
period0 is disabling mat5 and mat 1 at t=0.02
but at the same time, period1_2 is enabling mat5 back.

**Reply by ggb074113** [reply]
Thanks for your reply, I changed the input file slightly so that mat1 material is still used at the last time step, but I still can't understand why mat4 and mat5 always change in sync. What I didn't expect was that mat4 and mat5 were always enabled between 0 and 0.4s, and correctly disabled after 0.4s. The output at the last time step of 0.5s before I change the input file shows mat4 and mat5 enabled, and mat1 disabled, which is the same as the material enabled at time 0 to 0.4s. Is this caused by my period0 and period1_2 not working and mat5 being strangely set to turn on by default? Looking forward to your reply, thank you.
![Ubuntu 64 位-2024-12-16-17-05-22](https://github.com/user-attachments/assets/a13d489d-9c7a-4b5c-9d62-bef075b381b5)

**Reply by GiudGiud** [reply]
>  What I didn't expect was that mat4 and mat5 were always enabled between 0 and 0.4s, and correctly disabled after 0.4

Do you have a log showing this? This is not what I expect from the snippet

Note that should copy paste the input snippets rather than paste screenshots.

**Reply by ggb074113** [reply]
[testtimeperiod.zip](https://github.com/user-attachments/files/18161743/testtimeperiod.zip)
I followed https://github.com/idaholab/moose/blob/next/test/tests/controls/time_periods/materials/material.i and wrote the above file according to my problem for testing, and found that the same problem occurred in  this problem, is there something wrong with the way I write the input file or is it due to other reasons? The details of the output are as follows：
[testtimeperiod.zip](https://github.com/user-attachments/files/18161842/testtimeperiod.zip)

**Reply by GiudGiud** [reply]
Can you please use a public git repo instead of a zip file?

**Reply by ggb074113** [reply]
I'm sorry I'm not very good at using GitHub, which has caused you inconvenience. Just now, I did a simple test based on a situation similar to my problem according to https://github.com/idaholab/moose/blob/next/test/tests/controls/time_periods/materials/material.i. My input file and results are as follows: https://github.com/ggb074113/timeperiod_test-3times_matChange.git

**Reply by GiudGiud** [reply]
no worries.
 Sorry for the delay, I was traveling

you need to add `  reverse_on_false = false` to each TimePeriod. This makes sure they dont all "act" outside of the exact times they turn on/off
